### PR TITLE
Convert README to overview with list of links

### PR DESCRIPTION
**Pull request by davidjwiner**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -386,144 +164,5 @@ or connecting with our team in the `#ksp` channel in the
 Here are some planned features that have not yet been implemented:
 
 * Passing options to processors in Gradle.
-* Incremental processing.
 * Improved support for reading Java source code.
```
Comment (neetopia) [inline README.md:167]: Off topic: @ting-yuan I think we can remove these 2 items now since they are supported.

File: README.md [hunk 2]
```diff
@@ -165,214 +138,19 @@ class HelloFunctionFinderProcessor : SymbolProcessor() {
...
-* We are still investigating how to provide interoperability between processors.
-
-## For Java Annotation Processor Authors
-
-If you are familiar with the Java Annotation Processing API, see the
-[Java annotation processing to KSP reference](docs/reference.md) for how to
-implement most functions and data structures that are provided by Java's
-API.
+* [Quickstart](/docs/quickstart.md)
+* [Why use KSP?](/docs/why-ksp.md)
```
Comment (neetopia) [inline README.md:146]: Maybe we can put this in root as well?

**Review by neetopia — APPROVED** [review]
LGTM, just one minor quesiton.

File: docs/ksp-additional-details.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# How KSP models Kotlin code
+
+The API definition can be found [here](api/src/main/kotlin/com/google/devtools/ksp/).
```
Comment (ting-yuan) [inline docs/ksp-additional-details.md:3]: ../api/src/main/kotlin/com/google/devtools/ksp/

File: docs/ksp-additional-details.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+The diagram below is an overview of how Kotlin is [modeled](api/src/main/kotlin/com/google/devtools/ksp/symbol/) in KSP:
```
Comment (ting-yuan) [inline docs/ksp-additional-details.md:4]: ../api/src/main/kotlin/com/google/devtools/ksp/symbol/

File: docs/ksp-additional-details.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+![class diagram](../ClassDiagram.svg)
```
Comment (ting-yuan) [inline docs/ksp-additional-details.md:5]: maybe move the svg into docs as well.

File: README.md (hunk 2 above)
Comment (davidjwiner) [reply README.md:146]: Like put the text directly in the README? Or add those as separate markdown files in the root directory?

Comment (neetopia) [reply README.md:146]: Yeah, I mean put text directly in README.

File: README.md (hunk 1 above)
Comment (ting-yuan) [reply README.md:167]: Yes we do. Let's remove it in another PR. I'd like to remove "Migrating from old KSP releases"

File: README.md (hunk 2 above)
Comment (ting-yuan) [reply README.md:146]: The why-ksp paragraphs might be a little bit wordy so as a trade-off, would it make sense to mention the link in the introduction, in addition to being just an entry in the Resources section?

File: docs/ksp-additional-details.md (hunk 5 above)
Comment (davidjwiner) [reply docs/ksp-additional-details.md]: Done

File: docs/ksp-additional-details.md (hunk 4 above)
Comment (davidjwiner) [reply docs/ksp-additional-details.md]: Updated

File: docs/ksp-additional-details.md (hunk 3 above)
Comment (davidjwiner) [reply docs/ksp-additional-details.md]: Thanks for the catch, updated!

File: README.md (hunk 2 above)
Comment (davidjwiner) [reply README.md:146]: I added links to these two resources to the opening section as well (per Ting-Yuan's suggestion)

**Comment by ting-yuan** [comment]
Let's merge this and address the `why ksp` discussion in another PR.

**Pull request merged**